### ISSUE TITLE
Signed builds should be warning free

### ISSUE THREAD

**Issue by davkean** [dotnet/roslyn#8631]
We currently have the following warnings in the signed build, these should be cleaned up:

```
Found conflicts between different versions of the same dependent assembly that could not be resolved. These reference conflicts are listed in the build log when log verbosity is set to detailed.
Found conflicts between different versions of the same dependent assembly that could not be resolved. These reference conflicts are listed in the build log when log verbosity is set to detailed.
Could not resolve this reference. Could not locate the assembly "System.Collections.Immutable, Version=1.1.36.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
Found conflicts between different versions of the same dependent assembly that could not be resolved. These reference conflicts are listed in the build log when log verbosity is set to detailed.
Found conflicts between different versions of the same dependent assembly that could not be resolved. These reference conflicts are listed in the build log when log verbosity is set to detailed.
Could not resolve this reference. Could not locate the assembly "System.Collections.Immutable, Version=1.1.36.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a, processorArchitecture=MSIL". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation errors.
Could not resolve this reference. Could not locate the assembly "Microsoft.VisualStudio.Shell.Interop.14.0". Check to make sure the assembly exists on disk. If this reference is required by your code, you may get compilation error
```

**Comment by jasonmalinowski** [comment]
Some progress made on this: the redirect warnings are now gone. There's still some work to do with MicroBuild v2 configuration, source indexing, and our Roslyn.sln generation.
